src/Approximations/template_directions.jl

```diff
@@ -431,34 +669,49 @@ User-defined template directions.
 
 ### Notes
 
-Since custom directions may not be bounded, boundedness can either be asserted
-in the constructor or it will be determined automatically by performing an
-experiment (more precisely, we overapproximate the unit ball in the infinity
-norm using the given directions).
+This struct is a wrapper type for a set of user-defined directions which are
+iterated over. It has fields for the list of directions, the set dimension,
+and (boolean) cache fields for the boundedness and normalization properties.
+The latter are checked by default upon construction.
+
+To check boundedness, we overapproximate the unit ball in the infinity
+norm using the given directions and check if the resulting set is bounded.
 
 The dimension will also be determined automatically, unless the empty vector is
 passed (in which case the optional argument `n` needs to be specified).
+
+## Examples
+
+Creating a template with box directions in dimension two:
+
+```jldoctest
+julia> dirs = CustomDirections([[1.0, 0.0], [-1.0, 0.0], [0.0, 1.0], [0.0, -1.0]])
+CustomDirections{Float64,Array{Float64,1}}(Array{Float64,1}[[1.0, 0.0], [-1.0, 0.0], [0.0, 1.0], [0.0, -1.0]], 2, true, true)
+
+julia> LazySets.Approximations.isbounding(dirs)
+true
+
+julia> LazySets.Approximations.isnormalized(dirs)
+true
+```
 """
-struct CustomDirections{N, VN<:AbstractVector{N}} <: AbstractDirections{N}
+struct CustomDirections{N, VN<:AbstractVector{N}} <: AbstractDirections{N, VN}
     directions::Vector{VN}
     n::Int
-    isbounding::Bool
-
-    CustomDirections{N, VN}(directions::Vector{VN};
-                            n::Int=determine_dimension(directions),
-                            isbounding::Bool=_isbounding(directions)
-                           ) where {N, VN<:AbstractVector{N}} =
-        new{N, VN}(directions, n, isbounding)
+    bounded::Bool
+    normalized::Bool
 end
 
-# convenience constructor
-CustomDirections(directions::Vector{VN};
-                 n::Int=determine_dimension(directions),
-                 isbounding::Bool=_isbounding(directions)
-                ) where {N, VN<:AbstractVector{N}} =
-    CustomDirections{N, VN}(directions; n=n, isbounding=isbounding)
+function CustomDirections(directions::Vector{VN},
+                          n::Int=_determine_dimension(directions);
+                          check_boundedness::Bool=true,
+                          check_normalization::Bool=true) where {N, VN<:AbstractVector{N}}
+    bounded = check_boundedness ? _isbounding(directions) : false
+    normalized = check_normalization ? _isnormalized(directions) : false
+    return CustomDirections(directions, n, bounded, normalized)
+end
```

Comment:
How do you create a template where you know that the directions are bounding?

Reply:
i would in that case just use the type's fields, `CustomDirections(directions, n, bounded, normalized)`, since that's anyways a rather "advanced" use case.